python/ray/experimental/sgd/pytorch/pytorch_trainer.py

```diff
@@ -62,15 +62,23 @@ def __init__(self,
             resources_per_replica = utils.Resources(
                 num_cpus=1, num_gpus=0, resources={})
 
+        if backend == "auto":
+            backend = "nccl" if resources_per_replica.num_gpus > 0 else "gloo"
+
         Runner = ray.remote(
             num_cpus=resources_per_replica.num_cpus,
             num_gpus=resources_per_replica.num_gpus,
             resources=resources_per_replica.resources)(PyTorchRunner)
 
+        def calc_batch_size(i):
+            if i < batch_size % num_replicas:
+                return batch_size // num_replicas + 1
```

Comment:
nit: I think this could cause some problems if the underlying impl required all batch sizes to be the same (and would probably just return "//" and log an error message)